### PR TITLE
Add option for Offering/Tithe.

### PR DESCRIPTION
**Pull request by franzejr**

### CODE REVIEW

File: app/models/invoice.rb [hunk 1]
```diff
@@ -45,7 +51,7 @@ def subtotal
     if line_items.any?
       line_items.offers.map(&:amount).sum
```
Comment (adamdill) [inline app/models/invoice.rb]: If applied_coupon? is a ? method, shouldn't it only respond with bool? maybe we should need to rename this method, as i'm unclear why applied_coupon?  can respond with the total amount paid. seems like that should be a different method.

also i don't love the if/else/ternary operator
maybe we could make the first line a return, like: 
`return line_items.offers.map(&:amount).sum if line_items.any?`
and simplify the whole thing?

File: features/buying_an_offer.feature [hunk 2]
```diff
@@ -12,6 +12,18 @@ Feature: Buying an Offer
     And I should not see the Make Payment button
     And I should see the apply coupon button
 
+  @javascript
+  Scenario: User gives money to an offer. Thithe or Offering.
```
Comment (adamdill) [inline features/buying_an_offer.feature]: Scenario: User pays money towards an offer that doesn't have an amount.

File: features/buying_an_offer.feature [hunk 3]
```diff
@@ -12,6 +12,18 @@ Feature: Buying an Offer
...
+    Given I am logged in with a Payment Method
+    And I am on an offer page that is a tithe or offering
+    Then I click on Purchase link
```
Comment (adamdill) [inline features/buying_an_offer.feature]: When I click on the Purchase link

File: features/buying_an_offer.feature [hunk 4]
```diff
@@ -12,6 +12,18 @@ Feature: Buying an Offer
...
+    And I am on the invoice page
+    And I fill out the amount I want to pay
+    And I choose the Payment Method
+    And I click Make Payment
+    And I should be redirected to the Invoice page
+    And I should see a message I just paid for this offer
```
Comment (adamdill) [inline features/buying_an_offer.feature]: Then I should be redirected to the Invoice page

File: features/buying_an_offer.feature [hunk 5]
```diff
@@ -12,6 +12,18 @@ Feature: Buying an Offer
     And I should not see the Make Payment button
     And I should see the apply coupon button
 
+  @javascript
+  Scenario: User gives money to an offer. Thithe or Offering.
+    Given I am logged in with a Payment Method
+    And I am on an offer page that is a tithe or offering
```
Comment (adamdill) [inline features/buying_an_offer.feature]: And I am on an offer page for an offer that doesn't have an amount

File: features/steps/common_steps/pages.rb [hunk 6]
```diff
@@ -9,6 +9,13 @@ module Pages
       expect(page).to have_content(offer_name)
     end
 
+    step 'I am on an offer page that is a tithe or offering' do
```
Comment (adamdill) [inline features/steps/common_steps/pages.rb]: I am on an offer page for an offer that doesn't have an amount

File: app/models/invoice.rb [hunk 7]
```diff
@@ -16,9 +16,15 @@ class Invoice < ApplicationRecord
...
-    UNPAID
+    unless offer.amount
+      return zero_transactions? ? UNPAID : PAID
+    end
+
+    if offer.amount
+      return PAID if balance_remaining.zero?
+      return PARTIALLY_PAID if balance_remaining < total
+      return UNPAID
+    end
```
[the remark this answers is not in the document]
Comment (franzejr) [reply app/models/invoice.rb]: Ok. It makes sense and I think it will make the code more readable.

File: app/models/invoice.rb (hunk 1 above)
Comment (franzejr) [reply app/models/invoice.rb]: Applied coupon returns a boolean!

**Comment by franzejr** [comment]
@adamdill I did everything you told me. I just didn't understand about `applied_coupon?` you mentioned.

Probably before merge we could just have a quick cak? Otherwise, this is ready to be merged.

**Comment by adamdill** [comment]
this looks good with that one tiny change

**Pull request merged**